src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -62,8 +62,14 @@ public boolean validate() {
                 if (contentAsMap != null && contentAsMap.containsKey("password")) {
                     final String password = (String) contentAsMap.get("password");
 
+                    // Password is not allowed to be empty if present.
+                    if (password != null && password.isEmpty()) {
```

Comment:
Is the empty password checked only if regex is configured ? I see lines L56-57 inserted back in.
```
&& regex != null
&& !regex.isEmpty()	           
```

Reply:
Yes, [XContentHelper.convertToMap()](https://github.com/opendistro-for-elasticsearch/security/blob/017ae5f4ec975c3324934ab58df5305297979ba7/src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java#L61) has dependencies on `regex != null && !regex.isEmpty()`. Without Line 56-57, it throws out [`NotXContentException`](https://github.com/opendistro-for-elasticsearch/security/blob/017ae5f4ec975c3324934ab58df5305297979ba7/src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java#L100) and makes the `validate()` return a false and respond bad request.

Reply:
Could you explain the dependency of `XContentHelper.convertToMap()` on regex and why it is throwing `NotXContentException ` ? 

We would still allow users to use empty passwords unless users specify regex. Why do we want this behavior instead of rejecting empty passwords for all scenarios.

Reply:
> Could you explain the dependency of `XContentHelper.convertToMap()` on regex and why it is throwing `NotXContentException ` ?
> 
> We would still allow users to use empty passwords unless users specify regex. Why do we want this behavior instead of rejecting empty passwords for all scenarios.

There is a easy way to verify the dependencies. If you comment out the line 56 - 57 and run UserApiTest.testUserApiForNonSuperAdmin(), [this test](https://github.com/opendistro-for-elasticsearch/security/blob/4ed9f558b93082b44e80d5991d9b67c253e9f7f2/src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/UserApiTest.java#L537) will get a bad response instead of a forbidden. If you check the log, you'll see the `NotXContentException`.

Reply:
Are you saying the test code needs to be fixed? If yes, please fix as needed. 
`XContentHelper.convertToMap()` is an elasticsearch function. I don't see how it can be dependent on the regex configuration of the security plugin

Reply:
It's not dependent on the regex configuation. `XContentHelper.convertToMap()` cannot map XContent correctly when `XContent` content contains a `null` field. In this case, without `regex != null`, the regex can be null, which makes the `XContentHelper.convertToMap()` throw out [`NotXContentException`](https://github.com/elastic/elasticsearch/blob/724b244d7a72e18b9bd184f4bcb9cfc1f4967a16/server/src/main/java/org/elasticsearch/common/compress/CompressorFactory.java#L49)

Reply:
Regex is not part of content hence it cannot interfere with `XContentHelper.convertToMap()`

Reply:
It is converting array PATCH array body with PUT request in the test `testUserApiForNonSuperAdmin` and is failing to convert array into map. Need to fix the tests

Reply:
Ok. That's the culprit. I fixed it.